### DISCUSSION TITLE
Baudrate änderbar für RS485 Schnittstelle..

### DISCUSSION THREAD

**Discussion by Tomi200469** [skippermeister/OpenDTU-OnBattery#14] [Ideas]
Hallo an den Programmierer, folgende Situation …

Habe mir eine Batterie ( revolt , Pearl ) gekauft, bei dieser kann man das BMS Protokoll ändern ..
nach langem suchen bin ich auf deine Version gestoßen und diese ist anscheinend die einzige die
eine Abfrage an das BMS sendet und die DTU nicht nur wartet das Daten ankommen.
Deine Version sendet aber mit 115200 Baud und das ist das einzige was ich im BMS nicht ändern 
kann und ich benötige 9600 Baud .
Habe jetzt mal mit dem Terminal mit geschrieben was du an das BMS sendest und das dann mit 
9600 Baud an das BMS gesendet und dieses Antwortet auch, dieses hab ich dann mir 115200
Baud an die DTU gesendet, die zeigt dann auch Daten der Batterie an, nur das Timing ist halt
per Hand nicht einfach.
Wäre es möglich per Schalter in der Software die Baudrate auf 9600 zu ändern ?
Aktuell benutzt ich das openDTU Fusion v2 ,Pylontech Protokoll ,habe mir zwar Platformio installiert, 
habe aber noch zu wenig Erfahrungen damit.

Grüße

**Comment by skippermeister** [comment]
Hallo  @Tomi200469,
Da kann ich Dir einen Parameter einbauen. Ich denke über das WebGui im Setup Menü für die Batterie sollte ich das reinbekommen.
Hab etwas Geduld, ich bin beruflich aktuell sehr eingespannt.

**Reply by skippermeister** [reply]
@Tomi200469 
Ich habe eine neues Release hochgeladen und erstellt.
Es ist ein Parameter für über RS232 oder RS485 angeschlossene Batterien im Batterie Setup Menü hinzugefügt. Es können jetzt die Baudraten 9600, 19200, 38400 und 115200 gewählt werden.
Und für JKBMS wurde im Batterie Setup Menü die Eingabe der "Empfohlenen Lade- und Entladeschlussspannung hinzugefügt. 

Funktionieren jetzt bei Dir die Kommunikation zu Deiner Batterie ?

**Comment by Tomi200469** [comment]
Hallo und vielen Dank für die schnelle Umsetzung.Die Kommunikation funktioniert jetzt, aber einige Daten scheinen nicht ganz zu stimmen.... aber die wichtigsten sind vorhanden. 
![image](https://github.com/user-attachments/assets/06ac17de-3feb-4b51-b9a3-160748862c97)

**Reply by skippermeister** [reply]
@Tomi200469 
Das sieht schon gut aus. Deine Batterie hat anscheinend andere Bit Informationen für Fehler und Warnungen.
Welche Kapazität hat Deine Batterie ? Die hat doch sicherlich mehr als 10 Ah? Das wären 500Wh. Falls die Batterie 100Ah, entsprechen 5kWh hat, dann liefert die Batterie die Kapazität mit 2 Stellen hinter dem Komma und nicht wie die Pylontech mit 3 Stellen.
Auch werden die Parameter "Recommended charge voltage limit" und "Recommended discharge voltage limit" sowie "Recommended charge current limit" nicht geliefert. 
Auch Parameter wie "Charging possible" scheinen zu fehlen.
Falls Du beabsichtigst mit meiner Software und einem Meanwell Ladegrät die Batterie automatisch laden zu wollen, wird das nicht funktionieren. Ich benötige für den Ladealgorithmus diese Parameter. 
Schau Dir mal die von der Batterie gelieferten Parameter "Min./Max charge temperature" und "Min./Max. discharge temperature" an. Ich würde die Batterie niemals unter 0°C laden oder entladen. Auch die maximale Temperature 60 bzw. 65°C wäre mir zu kritisch. Erst bei diesen Temperaturen wird die Batterie selbst abschalten.

Dann viel Spaß mit dem Aufbau
